sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-jedis/src/test/java/com/azure/messaging/eventhubs/checkpointstore/jedis/JedisRedisCheckpointStoreTests.java

```diff
@@ -0,0 +1,172 @@
+    private JedisPool jedisPool;
+    private JedisRedisCheckpointStore store;
+    private Jedis jedis;
+    private JsonSerializer jsonSerializer;
+    @BeforeEach
+    public void setup() {
+        jedisPool = mock(JedisPool.class);
+        jedis = mock(Jedis.class);
+        store = new JedisRedisCheckpointStore(jedisPool);
+        jsonSerializer = JsonSerializerProviders.createInstance(true);
+    }
+
+    @Test
+    public void testListCheckpoints() {
+        //arrange
+        Checkpoint checkpoint = new Checkpoint()
+            .setConsumerGroup("consumerGroup")
+            .setEventHubName("eventHubName")
+            .setFullyQualifiedNamespace("fullyQualifiedNamespace")
+            .setPartitionId("one")
+            .setSequenceNumber((long) 1);
+        Set<String> value = new HashSet<>();
+        List<String> list = new ArrayList<>();
+        value.add("fullyQualifiedNamespace/eventHubNamespace/consumerGroup/one");
+        byte[] bytes = jsonSerializer.serializeToBytes(checkpoint);
+        list.add(new String(bytes));
+        //act
+        when(jedisPool.getResource()).thenReturn(jedis);
+        when(jedis.smembers("fullyQualifiedNamespace/eventHubName/consumerGroup")).thenReturn(value);
+        when(jedis.hmget(eq("fullyQualifiedNamespace/eventHubNamespace/consumerGroup/one"),
+            eq("checkpoint"))).thenReturn(list);
+        //assert
+        StepVerifier.create(store.listCheckpoints("fullyQualifiedNamespace", "eventHubName", "consumerGroup"))
+            .assertNext(checkpointTest -> {
+                assertEquals("fullyQualifiedNamespace", checkpointTest.getFullyQualifiedNamespace());
+                assertEquals("eventHubName", checkpointTest.getEventHubName());
+                assertEquals("consumerGroup", checkpointTest.getConsumerGroup());
+            })
+            .verifyComplete();
+    }
+
+    @Test
+    public void testListCheckpointsEmptyList() {
+        //arrange
+        //act
+        when(jedisPool.getResource()).thenReturn(jedis);
+        when(jedis.smembers("//")).thenThrow(new IllegalArgumentException());
+        //assert
+        Assertions.assertThrows(IllegalArgumentException.class, () -> store.listCheckpoints("", "", ""));
+    }
+
+    @Test
+    public void testCheckpointKeyNotStored() {
+        //arrange
+        Set<String> value = new HashSet<>();
+        value.add("fullyQualifiedNamespace/eventHubNamespace/consumerGroup/one");
+        //act
+        when(jedisPool.getResource()).thenReturn(jedis);
+        when(jedis.smembers("fullyQualifiedNamespace/eventHubName/consumerGroup")).thenReturn(value);
+        when(jedis.hmget(eq("fullyQualifiedNamespace/eventHubNamespace/consumerGroup/one"),
+            eq("checkpoint"))).thenThrow(new NoSuchElementException());
+        //assert
+        Assertions.assertThrows(NoSuchElementException.class, () -> store.listCheckpoints("fullyQualifiedNamespace", "eventHubName", "consumerGroup"));
```

Comment:
listCheckpoints should return a Flux.... I would expect the assertion to be using StepVerify.assertError or something similar.